distributed/core.py

```diff
@@ -132,12 +132,12 @@ class Server(TCPServer):
     default_port = 0
 
     def __init__(self, handlers, max_buffer_size=MAX_BUFFER_SIZE,
-            connection_limit=512, **kwargs):
+            connection_limit=512, ssl_context_creator= create_ssl_context, **kwargs):
         self.handlers = assoc(handlers, 'identity', self.identity)
         self.id = str(uuid.uuid1())
         self._port = None
         self.rpc = ConnectionPool(limit=connection_limit)
-        super(Server, self).__init__(max_buffer_size=max_buffer_size, **kwargs)
+        super(Server, self).__init__(max_buffer_size=max_buffer_size, ssl_options=ssl_context_creator(), **kwargs)
```

Comment:
This indirection seems unnecessary.  If possible I'd like to stick as close as possible to the Tornado API here.  I don't want to invent and maintain new API unless it's very convenient.  Is this very convenient?

Instead of taking in a thing that creates a set of options when called perhaps we can instead just take in the set of options?  If your goal is to default to config/environment variables then perhaps we can use a value like `ssl_options=True`?

```python
def __init__(..., ssl_options=False, ...):
    
    if ssl_options is True:
        ssl_options = ... default_options ...
    ...
    super(Server, self).__init__(..., ssl_options=ssl_options)
```